### ISSUE TITLE
Improve routing and page loading

### ISSUE THREAD

**Issue by mjulstein** [Altinn/app-frontend-react#373]
### Description

After upgrading from React Router v5 to react router v6 there are some benefits that can be achieved in performance and readability:
Currently the application loads an Entrypoint component on root. It has a lot of branching for different states of the application, and it could benefit form being split up using the <Outlet /> node to render what's unique for each state.
Further there is a ProcessWrapper, and a Presentation component.

This will also make it possible to point to a single page in an App, which is very convenient in for example Stateless Apps, where starting from root should not be a must. Direct linking to a page in the app would be possible, or maybe even indexing with search-engines?

### Additional Information

Direct links to pages cannot be done until all but the API routes are pointing to the react router app from serverside - atleast when switching to BrowserRouter from HashRouter. #344

### Tasks

Refactor the routing and components starting from App.tsx and utilize Outlet to reduce rerendering and support direct links to sub-routes.
